pkg/operation/botanist/component/nodelocaldns/nodelocaldns.go

```diff
@@ -0,0 +1,557 @@
+					},
+				},
+				FSGroup: policyv1beta1.FSGroupStrategyOptions{
+					Rule: policyv1beta1.FSGroupStrategyRunAsAny,
+				},
+				HostNetwork: true,
+				HostPorts: []policyv1beta1.HostPortRange{
+					{
+						Min: int32(53),
+						Max: int32(53),
+					},
+					{
+						Min: prometheusPort,
+						Max: prometheusPort,
+					},
+				},
+				Privileged: true,
+				RunAsUser: policyv1beta1.RunAsUserStrategyOptions{
+					Rule: policyv1beta1.RunAsUserStrategyRunAsAny,
+				},
+				SELinux: policyv1beta1.SELinuxStrategyOptions{
+					Rule: policyv1beta1.SELinuxStrategyRunAsAny,
+				},
+				SupplementalGroups: policyv1beta1.SupplementalGroupsStrategyOptions{
+					Rule: policyv1beta1.SupplementalGroupsStrategyRunAsAny,
+				},
+				Volumes: []policyv1beta1.FSType{
+					"secret",
+					"hostPath",
+					"configMap",
+				},
+			},
+		}
+
+		clusterRole = &rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "gardener.cloud:psp:kube-system:node-local-dns",
+				Labels: map[string]string{
+					v1beta1constants.LabelApp: LabelValue,
+				},
+			},
+			Rules: []rbacv1.PolicyRule{
+				{
+					APIGroups:     []string{"policy", "extensions"},
+					ResourceNames: []string{"gardener.kube-system.node-local-dns"},
+					Resources:     []string{"podsecuritypolicies"},
+					Verbs:         []string{"use"},
+				},
+			},
+		}
+
+		roleBinding = &rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "gardener.cloud:psp:node-local-dns",
+				Namespace: metav1.NamespaceSystem,
+				Labels: map[string]string{
+					v1beta1constants.LabelApp: LabelValue,
+				},
+				Annotations: map[string]string{resourcesv1alpha1.DeleteOnInvalidUpdate: "true"},
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: rbacv1.GroupName,
+				Kind:     "ClusterRole",
+				Name:     clusterRole.Name,
+			},
+			Subjects: []rbacv1.Subject{{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      serviceAccount.Name,
+				Namespace: serviceAccount.Namespace,
+			}},
+		}
+
+		configMap = &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "node-local-dns",
+				Namespace: metav1.NamespaceSystem,
+			},
+			Data: map[string]string{
+				configDataKey: domain + `:53 {
+  errors
+  cache {
+          success 9984 30
+          denial 9984 5
+  }
+  reload
+  loop
+  bind ` + c.bindIP() + `
+  forward . ` + c.values.ClusterDNS + ` {
+          ` + c.forceTcpToClusterDNS() + `
+  }
+  prometheus :` + strconv.Itoa(prometheusPort) + `
+  health ` + nodeLocal + `:8080
+  }
+in-addr.arpa:53 {
+  errors
+  cache 30
+  reload
+  loop
+  bind ` + c.bindIP() + `
+  forward . ` + c.values.ClusterDNS + ` {
+          ` + c.forceTcpToClusterDNS() + `
+  }
+  prometheus :` + strconv.Itoa(prometheusPort) + `
+  }
+ip6.arpa:53 {
+  errors
+  cache 30
+  reload
+  loop
+  bind ` + c.bindIP() + `
+  forward . ` + c.values.ClusterDNS + ` {
+          ` + c.forceTcpToClusterDNS() + `
+  }
+  prometheus :` + strconv.Itoa(prometheusPort) + `
+  }
```

Comment:
```suggestion
}
```

Shouldn't this be rather like this?